### PR TITLE
CharacterList component created

### PR DESCRIPTION
**Pull request by Efekahya**
image
https://www.figma.com/file/SBPCNjmePIwJCzqqhXF9Y8/Rick-and-Morty---Digieggs-Assessment?node-id=1044%3A5860

### CODE REVIEW

File: src/components/CharacterList/index.tsx [hunk 1]
```diff
@@ -1,6 +1,43 @@
...
+}
+export default function CharacterList({
+  characters,
+  count,
+}: CharacterListProps) {
+  if (count !== -1) {
+    characters = characters.slice(0, count)
+  }
+  return (
+    <>
```
Comment (isatanis) [inline src/components/CharacterList/index.tsx]: unnecessary fragment

File: src/components/CharacterList/index.tsx [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+  characters: ICharacter[] //TODO Characterler objesi gelince character tipi ile değiştirilecek
+  count: number
+}
+export default function CharacterList({
+  characters,
+  count,
+}: CharacterListProps) {
+  if (count !== -1) {
+    characters = characters.slice(0, count)
+  }
```
Comment (isatanis) [inline src/components/CharacterList/index.tsx]: you shouldn't try to update a prop. if you need a new variable, just define it.

File: src/components/CharacterList/index.tsx [hunk 3]
```diff
@@ -1,6 +1,43 @@
...
+  id: number
+  gender: string
+  spacies: string
+  origin: string
+  type: string
+  image: string
+  location: string
+  episodes: object[]
+}
+interface CharacterListProps {
```
Comment (isatanis) [inline src/components/CharacterList/index.tsx]: we're adding `I` to beginning of the interfaces. change it to `ICharacterListProps`

File: src/components/CharacterList/styles.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.characterList {
+    display: inline-grid;
```
Comment (isatanis) [inline src/components/CharacterList/styles.scss]: use flex instead of grid.

File: src/components/CharacterList/index.tsx [hunk 5]
```diff
@@ -1,6 +1,43 @@
 import React from "react"
-import "./styles.scss"
```
Comment (isatanis) [inline src/components/CharacterList/index.tsx:2]: why did you remove the styles?

File: src/components/CharacterList/index.tsx [hunk 6]
```diff
@@ -1,6 +1,43 @@
...
+  characters,
+  count,
+}: CharacterListProps) {
+  if (count !== -1) {
+    characters = characters.slice(0, count)
+  }
+  return (
+    <>
+      <div className="characterList">
+        {characters.map((character) => (
```
Comment (isatanis) [inline src/components/CharacterList/index.tsx]: use destructuring if there is no complex things in the function.

**Review by isatanis — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 7]
```diff
@@ -1,11 +1,51 @@
...
+    name: string
+    id: number
+    gender: string
+    spacies: string
+    origin: string
+    type: string
+    image: string
+    location: string
+    episodes: object[]
+  }
```
Comment (bybash) [inline src/App.tsx]: Don't define interface or type inside the components. If you want to define them in the same file, define them at the top before the component definition. Also, you can define them in a separate type file and then export and use it wherever you want.

File: src/App.tsx [hunk 8]
```diff
@@ -1,11 +1,51 @@
...
+
+  const character: ICharacter = {
```
Comment (bybash) [inline src/App.tsx]: You can define them in a mock file and then you can import it and use it.

File: src/components/CharacterList/index.tsx [hunk 9]
```diff
@@ -1,6 +1,44 @@
...
+  gender: string
+  spacies: string
+  origin: string
+  type: string
+  image: string
+  location: string
+  episodes: object[]
+}
+interface ICharacterListProps {
+  characters: ICharacter[] //TODO Characterler objesi gelince character tipi ile değiştirilecek
```
Comment (bybash) [inline src/components/CharacterList/index.tsx]: Please write your comments in English.

File: src/components/CharacterList/index.tsx [hunk 10]
```diff
@@ -1,6 +1,44 @@
...
+  count: number
+}
+export default function CharacterList({
+  characters,
+  count,
+}: ICharacterListProps) {
+  let sliced
+  if (count !== -1) {
+    sliced = characters.slice(0, count)
+  }
```
Comment (bybash) [inline src/components/CharacterList/index.tsx]: Need an else block, otherwise sliced will be undefined.

File: src/index.tsx [hunk 11]
```diff
@@ -1,19 +1,17 @@
-import React from 'react';
-import ReactDOM from 'react-dom/client';
+import React from "react"
+import ReactDOM from "react-dom/client"
 import "./styles.scss"
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import App from "./App"
```
Comment (bybash) [inline src/index.tsx:3]: Try to order your imports.
Import order:
1) Third parties
2) Component
3) Utilities, helper functions
4) Images
5) Styles

File: src/index.tsx [hunk 12]
```diff
@@ -1,19 +1,17 @@
...
+import reportWebVitals from "./reportWebVitals"
```
Comment (bybash) [inline src/index.tsx]: I don't think we need reportWebVitals you can remove it.

File: src/styles.scss [hunk 13]
```diff
@@ -1,17 +1,18 @@
-@import url('https://fonts.googleapis.com/css2?family=Inter&display=swap');
+@import url("https://fonts.googleapis.com/css2?family=Inter&display=swap");
 @import "./styles/variables.scss";
 @import "./components/Navbar/styles.scss";
+@import "./components/CharacterList/styles.scss";
```
Comment (bybash) [inline src/styles.scss:5]: Import component styles inside components' own file, no need to import all of them here.

**Review by bybash — CHANGES_REQUESTED** [review]
Nice 👍 Added some comments.

**Review by bybash — APPROVED** [review]

**Review by iremlaya — APPROVED** [review]

**Pull request merged**